File: data/derived_cot/rq2_eval/translation/qwen/_converted_code/cpp_to_py/HRManagementSystem.py
class HRManagementSystem:
    def __init__(self):
        self.employees = {}

    def add_employee(self, employee_id, name, position, department, salary):
        if employee_id in self.employees:
            return False
        self.employees[employee_id] = {
            "name": name,
            "position": position,
            "department": department,
            "salary": salary
        }
        return True

    def remove_employee(self, employee_id):
        if employee_id not in self.employees:
            return False
        del self.employees[employee_id]
        return True

    def update_employee(self, employee_id, employee_info):
        if employee_id not in self.employees:
            return False
        emp_data = self.employees[employee_id]
        for key in employee_info:
            if key not in emp_data:
                return False
        emp_data.update(employee_info)
        return True

    def get_employee(self, employee_id):
        if employee_id in self.employees:
            return self.employees[employee_id]
        return None

    def list_employees(self):
        return {
            emp_id: {"employee_ID": emp_id, **emp_data}
            for emp_id, emp_data in self.employees.items()
        }

File: data/derived_cot/rq3_prompting/results/translation/qwen/py_to_cpp/code_output/HRManagementSystem.cpp
#include <string>
#include <map>
#include <optional>

class HRManagementSystem {
private:
    std::map<int, std::map<std::string, std::string>> employees;

public:
    // Constructor
    HRManagementSystem() = default;

    // Add an employee
    bool add_employee(int employee_id, const std::string& name, const std::string& position, const std::string& department, int salary) {
        if (employees.find(employee_id) != employees.end()) {
            return false;
        }
        std::map<std::string, std::string> emp_data;
        emp_data["name"] = name;
        emp_data["position"] = position;
        emp_data["department"] = department;
        emp_data["salary"] = std::to_string(salary);
        employees[employee_id] = emp_data;
        return true;
    }

    // Remove an employee
    bool remove_employee(int employee_id) {
        if (employees.find(employee_id) != employees.end()) {
            employees.erase(employee_id);
            return true;
        }
        return false;
    }

    // Update an employee
    bool update_employee(int employee_id, const std::map<std::string, std::string>& employee_info) {
        auto it = employees.find(employee_id);
        if (it == employees.end()) {
            return false;
        }
        const auto& emp = it->second;
        for (const auto& kv : employee_info) {
            std::string key = kv.first;
            if (key == "name") {
                emp[key] = kv.second;
            } else if (key == "position") {
                emp[key] = kv.second;
            } else if (key == "department") {
                emp[key] = kv.second;
            } else if (key == "salary") {
                emp[key] = kv.second;
            } else {
                return false;
            }
        }
        it->second = emp;
        return true;
    }

    // Get an employee
    std::optional<std::map<std::string, std::string>> get_employee(int employee_id) {
        auto it = employees.find(employee_id);
        if (it != employees.end()) {
            return it->second;
        }
        return std::nullopt;
    }

    // List employees
    std::map<int, std::map<std::string, std::string>> list_employees() {
        std::map<int, std::map<std::string, std::string>> employee_data;
        for (const auto& kv : employees) {
            int id = kv.first;
            const auto& emp = kv.second;
            std::map<std::string, std::string> emp_data;
            emp_data["employee_ID"] = std::to_string(id);
            emp_data["name"] = emp.at("name");
            emp_data["position"] = emp.at("position");
            emp_data["department"] = emp.at("department");
            emp_data["salary"] = emp.at("salary");
            employee_data[id] = emp_data;
        }
        return employee_data;
    }
};